components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApiCategoriesApiServiceImpl.java

```diff
@@ -43,21 +44,21 @@
     private static final Log log = LogFactory.getLog(ApiCategoriesApiServiceImpl.class);
 
     public Response apiCategoriesGet(String xWSO2Tenant, MessageContext messageContext) {
-        String requestedTenantDomain = RestApiUtil.getRequestedTenantDomain(xWSO2Tenant);
+        String organization = (String) messageContext.get(RestApiConstants.ORGANIZATION);
         try {
-            if (!APIUtil.isTenantAvailable(requestedTenantDomain)) {
-                RestApiUtil.handleBadRequest("Provided tenant domain '" + xWSO2Tenant + "' is invalid",
+            if (!APIUtil.isOrganizationTenantAvailable(organization)) {
+                RestApiUtil.handleBadRequest("Tenant for the provided organization '" + organization + "' is invalid",
                         ExceptionCodes.INVALID_TENANT.getErrorCode(), log);
```

Comment:
if (!APIUtil.isTenantAvailable(requestedTenantDomain)) {} is not needed. this check is done from the interceptor level

Reply:
Done via https://github.com/wso2/carbon-apimgt/pull/10539/commits/d9368dcf8df1c26ded2c9ff81b077a5267d30044